### PR TITLE
Add method for searching cases by the new envelope id field

### PR DESCRIPTION
**Pull request by rkondratowicz**
https://tools.hmcts.net/jira/browse/BPS-1339

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CcdApi.java [hunk 1]
```diff
@@ -155,6 +155,21 @@ public CaseDetails getCase(String caseRef, String jurisdiction) {
...
+        ServiceConfigItem serviceConfig = serviceConfigProvider.getConfig(service);
+        return searchCases(
+            serviceConfig.getJurisdiction(),
+            String.join(",", serviceConfig.getCaseTypeIds()),
+            "      {\n"
+                + "  \"query\": {\n"
+                + "    \"term\" : {\n"
+                + "      \"data.bulkScanEnvelopes.value.id\" : \"" + envelopeId + "\"\n"
+                + "    }\n"
+                + "  }\n"
```
Comment (lgonczar) [inline src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CcdApi.java:168]: I guess these `\n` aren't necessary.

File: src/test/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CcdApiSearchTest.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    @Mock CoreCaseDataApi feignCcdApi;
+    @Mock CcdAuthenticatorFactory authenticatorFactory;
+    @Mock ServiceConfigProvider serviceConfigProvider;
+
+    @Mock CcdAuthenticator ccdAuthenticator;
+
+    ArgumentCaptor<String> idamTokenCaptor = ArgumentCaptor.forClass(String.class);
+    ArgumentCaptor<String> s2sTokenCaptor = ArgumentCaptor.forClass(String.class);
+    ArgumentCaptor<String> caseTypeCaptor = ArgumentCaptor.forClass(String.class);
+    ArgumentCaptor<String> searchQueryCaptor = ArgumentCaptor.forClass(String.class);
```
Comment (lgonczar) [inline src/test/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CcdApiSearchTest.java:33]: I'd put these captors in the test, as they seem to belong to it. Up to you.

File: src/test/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CcdApiSearchTest.java [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+        // then
+        assertThat(caseRefs).containsExactly(111L, 222L);
+
+        assertThat(idamTokenCaptor.getValue()).isEqualTo(idamToken);
+        assertThat(s2sTokenCaptor.getValue()).isEqualTo(s2sToken);
+        assertThat(caseTypeCaptor.getValue()).isEqualTo("a,b"); //concatenated case types for current service
+        assertThat(searchQueryCaptor.getValue()).contains(envelopeId);
+    }
+
+    private ServiceConfigItem cfg(String jurisdiction, List<String> caseTypes) {
```
Comment (lgonczar) [inline src/test/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CcdApiSearchTest.java:88]: `cfg` is fine within the method, but maybe we should give the method itself a full name?

File: src/test/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CcdApiSearchTest.java (hunk 2 above)
Comment (rkondratowicz) [reply src/test/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CcdApiSearchTest.java:33]: Before this test class were added there were not unit tests for CcdApi.
Hopefully more test will be added to this class soon and those captors will be used as well.

File: src/test/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CcdApiSearchTest.java (hunk 3 above)
Comment (rkondratowicz) [reply src/test/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CcdApiSearchTest.java]: fixed

**Comment by rkondratowicz** [comment]
Broken jenkins pipeline, reopening

**Pull request closed without merging**